tesseract/tesseract_planning/tesseract_command_language/include/tesseract_command_language/command_language_utils.h

```diff
@@ -138,64 +138,42 @@ long getPlanInstructionsCount(const CompositeInstruction& composite_instruction,
  */
 const Eigen::VectorXd& getJointPosition(const Waypoint& waypoint);
 
-/**
- * @brief Helper function used by Flatten. Not intended for direct use
- * @param flattened Vector of instructions representing the full flattened composite
- * @param composite Composite instruction to be flattened
- * @param include_composite If true, CompositeInstructions will be included in the final flattened vector
- */
-void flattenHelper(std::vector<std::reference_wrapper<Instruction>>& flattened,
-                   CompositeInstruction& composite,
-                   const bool& process_child_composites);
+/** @brief This is used for filtering only what you want in the vector **/
+using flattenFilter = std::function<bool(const Instruction&)>;
 
 /**
  * @brief Flattens a CompositeInstruction into a vector of Instruction&
- * @param instruction Input composite instruction to be flattened
- * @param include_composite Default: false. If true, CompositeInstructions will be included in the final flattened
- * vector
+ *
+ * If /p composite_instruction parameter has a start instruction it is added but child composites are not check for
```

Comment:
I tried to post this earlier but I think it got lost somehow.

This seems like a somewhat not readily apparent logic to have in a flatten utility. This logic is a bit specific to the planning. Some processes might not want the start instruction and then they have to know to remove it. Others might want to include all of the start instructions.

For reference, the start instruction was the reason I added the include composite flag. That includes a reference to the composite itself so the caller can check if there is a start instruction. Then you get to decide if you use only the first, all l, or none of them.

Reply:
I am restructuring these utility function to make them generic where you can provide a function callback to indicate if something should be kept or not. Then I will create one that are specific to motion planning in the planning utilities.

Reply:
@mpowelson Take a look at how I restructured the command language utilities and let me know what you think.

Reply:
That looks good. The unit tests for the filter functionality look a bit sparse. It only tests when it returns true. I imagine we want to test the default value and return false (and includes only the correct types). 

Should we add some extra information about what exactly those filters should do? e.g. They should return true when the instruction passed in should be included, and they should not throw. Including some premade ones would be convenient. I'm thinking one to include all planning instructions, all non-user-implemented, etc.